components/datadog/dogstatsd-standalone/k8s.go

```diff
@@ -0,0 +1,273 @@
+	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
+	v1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/rbac/v1"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+// HostPort defines the port used by the dogstatsd standalone deployment. The
+// standard port for dogstatsd is 8125, but it's already used by the agent.
+const HostPort = 8128
+
+// Socket defines the socket exposed by the dogstatsd standalone deployment.
+// It's not the default to avoid conflict with the agent.
+const Socket = "/var/run/datadog/dsd-standalone.socket"
+
+type K8sComponent struct {
+	pulumi.ResourceState
+}
+
+func K8sAppDefinition(e config.CommonEnvironment, kubeProvider *kubernetes.Provider, namespace string, fakeIntake *ddfakeintake.ConnectionExporter, kubeletTLSVerify bool, opts ...pulumi.ResourceOption) (*K8sComponent, error) {
+	opts = append(opts, pulumi.Provider(kubeProvider), pulumi.Parent(kubeProvider), pulumi.DeletedWith(kubeProvider))
+
+	k8sComponent := &K8sComponent{}
+	if err := e.Ctx.RegisterComponentResource("dd:dogstatsd-standalone", "dogstatsd", k8sComponent, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(k8sComponent))
+
+	ns, err := corev1.NewNamespace(
+		e.Ctx,
+		namespace,
+		&corev1.NamespaceArgs{
+			Metadata: metav1.ObjectMetaArgs{
+				Name: pulumi.String(namespace),
+			},
+		},
+		opts...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, utils.PulumiDependsOn(ns))
+
+	envVars := corev1.EnvVarArray{
+		&corev1.EnvVarArgs{
+			Name:  pulumi.String("DD_API_KEY"),
+			Value: e.AgentAPIKey(),
+		},
+		&corev1.EnvVarArgs{
+			Name: pulumi.String("DD_KUBERNETES_KUBELET_HOST"),
+			ValueFrom: corev1.EnvVarSourceArgs{
+				FieldRef: corev1.ObjectFieldSelectorArgs{
+					FieldPath: pulumi.String("status.hostIP"),
+				},
+			},
+		},
+		&corev1.EnvVarArgs{
+			Name:  pulumi.String("DD_DOGSTATSD_NON_LOCAL_TRAFFIC"),
+			Value: pulumi.StringPtr("true"),
+		},
+		&corev1.EnvVarArgs{
+			Name:  pulumi.String("DD_DOGSTATSD_ORIGIN_DETECTION"),
+			Value: pulumi.StringPtr("true"),
+		},
+		&corev1.EnvVarArgs{
+			Name:  pulumi.String("DD_DOGSTATSD_SOCKET"),
+			Value: pulumi.StringPtr(Socket),
+		},
+		&corev1.EnvVarArgs{
+			Name:  pulumi.String("DD_DOGSTATSD_TAG_CARDINALITY"),
+			Value: pulumi.StringPtr("orchestrator"),
+		},
+		&corev1.EnvVarArgs{
+			Name:  pulumi.String("DD_KUBELET_TLS_VERIFY"),
+			Value: pulumi.String(strconv.FormatBool(kubeletTLSVerify)),
+		},
+	}
+
+	if fakeIntake != nil {
+		envVars = append(
+			envVars,
+			&corev1.EnvVarArgs{
+				Name:  pulumi.String("DD_ADDITIONAL_ENDPOINTS"),
+				Value: pulumi.Sprintf(`{"http://%s": ["FAKEAPIKEY"]}`, fakeIntake.Host),
+			},
+			&corev1.EnvVarArgs{
+				Name:  pulumi.String("DD_LOGS_CONFIG_ADDITIONAL_ENDPOINTS"),
+				Value: pulumi.Sprintf(`[{"host": "%s", "port": 80, "is_reliable": true, "usessl": false}]`, fakeIntake.Host),
+			},
```

Comment:
This one shouldn’t be needed as `dogstatsd-standalone` won’t run a logs agent.

Reply:
Right, this was a copy-pasta oversight.